### ISSUE TITLE
Switch to Hatch

### ISSUE THREAD

**Issue by nstarman** [astropy/astropy-iers-data#25]
As the build system

**Comment by astrofrog** [comment]
Just curious, what are the benefits and why specifically for this package?

**Comment by nstarman** [comment]
https://scientific-python-cookie.readthedocs.io/en/latest/guides/packaging-simple/ has a good overview of all the tools. Hatch is recommended for pure python packages, like this one. It has good support for dynamic versioning and allows us to get rid of the Manifest.in. It's a super easy swap (couple lines in pyproject.toml).

**Comment by SauravMaheshkar** [comment]
The release automation might also need to be updated in case we proceed with this, right?

**Comment by astrofrog** [comment]
@neutrinoceros - if we did switch build system, is it correct that your recommendation is to use flit and not hatch?

**Comment by neutrinoceros** [comment]
> I'm assuming for this whole thread that you mean `hatchling` instead of `hatch`, and `flit-core` instead of `flit`.
> `hatch` and `flit` are build frontends, that any developer may choose to use individually, while `hatchling` and `flit-core` are build backends, which is to be set on the package itself.


yes, but `hatchling` is a perfectly fine choice in my book too. My main reason to recommend `flit-core` over `hatchling` is that it doesn't try to do as much so I *expect* it to be more stable, so it may require (even) less maintenance, but that is a bet.

**Comment by nstarman** [comment]
Not a strong opinion but see the VCS compat for why I tend to use hatch. 
https://learn.scientific-python.org/development/guides/packaging-simple/#includingexcluding-files-in-the-sdist
And dynamic versioning.

**Comment by SauravMaheshkar** [comment]
> [scientific-python-cookie.readthedocs.io/en/latest/guides/packaging-simple](https://scientific-python-cookie.readthedocs.io/en/latest/guides/packaging-simple/) has a good overview of all the tools. Hatch is recommended for pure python packages, like this one. It has good support for dynamic versioning and allows us to get rid of the Manifest.in. It's a super easy swap (couple lines in pyproject.toml).

Not sure if we can get rid of the `MANIFEST.in` by switching to hatch. As per the docs at [scientific-python.org](https://learn.scientific-python.org/development/guides/packaging-simple/#includingexcluding-files-in-the-sdist) hatchling ignores as per the gitignore. At present the `MANIFEST.in` contains the following files

```
# All files which are tracked by git and not explicitly excluded here are included by setuptools_scm
prune .github
exclude .gitignore
exclude scripts/update_data.sh
exclude MANIFEST.in
```

All these files can't be added to the gitignore

**Comment by neutrinoceros** [comment]
It's actually additive. Files can be selected via configuration too: https://hatch.pypa.io/latest/config/build/#artifacts

**Comment by SauravMaheshkar** [comment]
> It's actually additive. Files can be selected via configuration too: [hatch.pypa.io/latest/config/build#artifacts](https://hatch.pypa.io/latest/config/build/#artifacts)

Ahh nice didn't know that. Nice catch !
